### PR TITLE
[Bug] Fix issue with register_button function

### PR DESCRIPTION
**Pull request by drashna**
Specifically, it's not enough to just set the report here, you need to also send the report immediately.  

It's making the assumption that every `pointing_device_task()` call will send the report to the host system.  Normally, that would happen. But doing that keeps the host system awake. The best/proper way to handle this is to immediately send the report after it's been changed.

I did not catch this until testing and user feedback. 

## Types of Changes

- [x] Core
- [x] Bugfix

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
@manna-harbour I didn't catch this with your code change, as I didn't realize it was a problem (that's on me).  But in testing my dactyl manuform with a trackball, I started running into this issue, and force sending the report fixes the issue 

Just wanted to let you know/flag you for verification.

**Review by fauxpark — APPROVED** [review]

**Comment by manna-harbour** [comment]
I deliberately didn't do that so that the buttons can be manipulated in `pointing_device_task()` if desired, say to implement middle button scroll, and the default `pointing_device_task()` is just `pointing_device_send()` anyway.  Are you saying the default `pointing_device_task()` keeps the host awake and needs to be changed?  In any case, I only added this in preparation for a future build, I'm not planning on doing button manipulation, and middle button scroll could always be added as a feature later if anyone actually wants it.

**Comment by drashna** [comment]
Yeah, it's a tricky situation, I think. 
If you always send the report, it will keep the system awake, as it's considered mouse movement, even if the report doesn't change.  

Because of this, I only send the report if the report changes.   However, with this, as is, hitting the mouse keys isn't considered a change, so it doesn't send.  That means that it waits  until an actual report happens, and this happens with mouse keys disabled, before I added the "send" in keyboard code. 
Though, the reason that it doesn't detect the change, is because the task runs the "get" function, runs the sensor code, and then compares.  It may be possible to handle this differently, but IMO, it shouldn't be on the keyboard/user code to handle that.

Basically, this should be sent immediately, IMO.  However, we could hide it behind a define. 


The other option would be to move the check to `pointing_device_send` and don't send the report to the host (eg, don't call `host_mouse_send`) unless the report has changed. 

I'm not sure which is better, but most everywhere else, the report is immediately sent after the report has been changed, I feel that this is probably the best way to handle this.

**Comment by manna-harbour** [comment]
`pointing_device_send()` needs to be removed from the default `pointing_device_task()` so the host doesn't see continuous mouse movement.  I don't see that change here.  Is there another unlinked PR?  Also, the few user implementations of `pointing_device_task()` should also be changed to only `call pointing_device_send()` on change.

> The other option would be to move the check to `pointing_device_send` and don't send the report to the host (eg, don't call `host_mouse_send`) unless the report has changed.
> 
> I'm not sure which is better, but most everywhere else, the report is immediately sent after the report has been changed, I feel that this is probably the best way to handle this.

When `mousekey` buttons are used with `ps2_mouse` the report is not sent immediately so that the buttons can be manipulated there, in particular to support middle button scroll.  Sending immediately for `pointing_device` would mean buttons can not be manipulated in user `pointing_device_task()`, so I'd say it'd be better to add the check to `pointing_device_send()` instead.  That would also mean user implementations of `pointing_device_task()` won'd need to do the check.  In practice though it won't make any difference to me either way.

**Comment by drashna** [comment]
Well, for pointing device, the `send` has to be done manually, and is done so by the task function. Which means either blindly sending it, or adding the checks.   

I have some code in a branch, but not a PR, since it's definitely not ready. 


For now, I think that something like this may be best: 
```c
    pointing_device_set_report(currentReport);
#        ifndef PS2_MOUSE_ENABLE
    pointing_device_send();
#        endif 
#    endif
```

**Comment by manna-harbour** [comment]
By adding the check to `pointing_device_send()` I actually meant to store the last sent mouse report and only call `host_mouse_send()` if it has changed.

**Comment by drashna** [comment]
I know, but I'm hesitant to do so for now, since it may make having the keyboard waking on mouse movement harder to code.   I'd rather have a simple change now, and then implement a better version and revert this (if needed) later.    And since I'm using the pointing device code as my daily driver for a mouse, I *will* be working on that.

**Comment by drashna** [comment]
In an unrelated issue, mousekeys does basically the same thing that I'm proposing, actually. And it has some weird issues, at a low level.  

And I think "fix" for that would be the same as the more complicated solution to this, that I have been wanting to avoid.

**Comment by drashna** [comment]
I'm closing this. Will open up a varient of this that actually detects changes, as that will be the better way to go.
